gson/src/main/java/com/google/gson/TypeAdapter.java

```diff
@@ -216,7 +216,7 @@ public final String toJson(T value) {
     try {
       toJson(stringWriter, value);
     } catch (IOException e) {
-      throw new AssertionError(e); // No I/O writing to a StringWriter.
+      throw new JsonIOException(e);
```

Comment:
I guess this means there should be `@throws JsonIOException` in the javadoc?

Reply:
Have also added it for the other `TypeAdapter` methods wrapping `IOException`s. I hope the wording is alright.